plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -1059,9 +1059,7 @@ private function maybe_sync_order( \WC_Abstract_Order &$order, \WC_Abstract_Orde
 		 *
 		 * So we write back to the order table when order modified date is more recent than post modified date. Otherwise, we write to the post table.
 		 */
-		if ( $order_modified_date > $post_order_modified_date ) {
-			return;
-		} else {
+		if ( $post_order_modified_date > $order_modified_date ) {
```

Comment:
If we're just simplifying the code down to keep things nice and clean, this would maintain the previous behaviour (in which we would also call `migrate_post_record` when the modification dates are equal):

```suggestion
		if ( $post_order_modified_date >= $order_modified_date ) {
```

This also fixes the three `OrdersTableDataStoreTests` fails.

Reply:
I neglected the tests, my fault. But the removal of the '=' comparison is on purpose: if both records have the exact same update date it means that they are in sync and thus there is no need to migrate, right?

Reply:
Due to the check at the top of the method [(is the post different from the order?)](https://github.com/woocommerce/woocommerce/pull/35125/files#diff-8c34c1a6f743002d1f2db13c299d2086d0209889f72abc86bef3bd734abb2585R1044-R1046) I assumed that if we reach this point then there *is* some kind of difference between the two records that ought to be reconciled?

Reply:
After discussing with @vedanshujain I'm applying this change. Turns out that data comparison is needed too when the dates are equal, since data in the metadata table could have been modified as well.